docs/site/hasOne-relation.md

```diff
@@ -0,0 +1,270 @@
+contain a custom foreign key to be stored as the relation metadata. The
+decorator logic also designates the relation type and tries to infer the foreign
+key on the target model (`keyTo` in the relation metadata) to a default value
+(source model name appended with `id` in camel case, same as LoopBack 3).
+
+The decorated property name is used as the relation name and stored as part of
+the source model definition's relation metadata.
+
+A usage of the decorator with a custom foreign key name for the above example is
+as follows:
+
+```ts
+// import statements
+class Supplier extends Entity {
+  // constructor, properties, etc.
+  @hasOne(() => Account, {keyTo: 'suppId'})
+  account?: Account;
+}
+```
+
+## Setting up your database for hasOne relation - MySQL
+
+At the moment, LoopBack does not provide the means to enforce referential
+integrity for the `hasOne` relation. It is up to users to set this up at the
+database layer so constraints are not violated. Let's take MySQL as the backing
+database for our application. Given the `supplier` has one `account` scenario
+above, we need to run two SQL statements on the `account` table for the database
+to enforce referential integrity and align with LoopBack's `hasOne` relation.
+
+1. Make `supplierId` property or column a foreign key which references the `id`
+   from Supplier model's `id` property:
+
+```sql
+ALTER TABLE <databaseName>.Account ADD FOREIGN KEY (supplierId) REFERENCES <databaseName>.Supplier(id);
+```
+
+2. Create a unique index for the same property `supplierId`, so that for each
+   supplier instance, there is only one associated account instance.
+
+```sql
+   ALTER TABLE <databaseName>.Account ADD UNIQUE INDEX supplierIndex (supplierId);
+```
+
+Before making the following changes, please follow the steps outlined in
+[Database Migrations](Database-migrations.md) to create the database schemas
+defined by the models in your application.
+
+## Configuring a hasOne relation
+
+The configuration and resolution of a `hasOne` relation takes place at the
+repository level. Once `hasOne` relation is defined on the source model, then
+there are a couple of steps involved to configure it and use it. On the source
+repository, the following are required:
+
+- In the constructor of your source repository class, use
+  [Dependency Injection](Dependency-injection.md) to receive a getter function
+  for obtaining an instance of the target repository. _Note: We need a getter
+  function, accepting a string repository name instead of a repository
+  constructor, or a repository instance, in account to break a cyclic dependency
+  between a repository with a hasOne relation and a repository with the matching
+  belongsTo relation._
+
+- Declare a property with the factory function type
+  `HasOneRepositoryFactory<targetModel, typeof sourceModel.prototype.id>` on the
+  source repository class.
+- call the `createHasOneRepositoryFactoryFor` function in the constructor of the
```

Comment:
nitpick: `Call`

Reply:
createHasOneRepositoryFactoryFor does not exist.  For some reason _createHasOneRepositoryFactoryFor does.  Is this correct?